src/atomate2/common/flows/anharmonicity.py

```diff
@@ -0,0 +1,276 @@
+            Seed to use for the random number generator
+            (only used if one_shot_approx == False)
+        atom_resolved: bool
+            If true, calculate the atom-resolved sigma^A. This is false by default.
+        mode_resolved: bool
+            If true, calculate the mode-resolved sigma^A. This is false by default.
+        n_samples: int
+            Number of times displaced structures are sampled.
+            Must be >= 1 and cannot be used when one_shot_approx == True.
+
+        Returns
+        -------
+        Flow
+            The workflow for the anharmonicity calculations
+        """
+        phonon_flow = self.phonon_maker.make(
+            structure,
+            prev_dir,
+            born,
+            epsilon_static,
+            total_dft_energy_per_formula_unit,
+            supercell_matrix,
+        )
+
+        phonon_doc = phonon_flow.output
+        anharmon_flow = self.make_from_phonon_doc(
+            phonon_doc,
+            prev_dir,
+            temperature,
+            one_shot_approx,
+            seed,
+            atom_resolved,
+            mode_resolved,
+            n_samples,
+        )
+
+        results = store_results(
+            sigma_dict=anharmon_flow.output,
+            phonon_doc=phonon_flow.output,
+            one_shot=one_shot_approx,
+        )
+
+        jobs = [phonon_flow, anharmon_flow, results]
+        return Flow(jobs, results.output)
+
+    def make_from_phonon_doc(
+        self,
+        phonon_doc: PhononBSDOSDoc,
+        prev_dir: str | Path | None = None,
+        temperature: float = 300,
+        one_shot_approx: bool = True,
+        seed: int | None = None,
+        atom_resolved: bool = False,
+        mode_resolved: bool = False,
+        n_samples: int = 1,
+    ) -> Flow:
+        """Create an anharmonicity workflow from a phonon calculation.
+
+        Parameters
+        ----------
+        phonon_doc: PhononBSDOSDoc
+            The document to get the anharmonicity for
+        prev_dir : str or Path or None
+            A previous calculation directory to use for copying outputs.
+        temperature: float
+            The temperature for the anharmonicity calculation
+        one_shot_approx: bool
+            If true, finds the one shot approximation of sigma^A and if false,
+            finds the full sigma^A. The default is True.
+        seed: int | None
+            Seed to use for the random number generator
+            (only used if one_shot_approx == False)
+        atom_resolved: bool
+            If true, calculate the atom-resolved sigma^A. This is false by default.
+        mode_resolved: bool
+            If true, calculate the mode-resolved sigma^A. This is false by default.
+        n_samples: int
+            Number of times displaced structures are sampled.
+            Must be >= 1 and cannot be used when one_shot_approx == True.
+        """
+        if phonon_doc.has_imaginary_modes:
+            warn(
```

Comment:
Potentially add a comment here stating that this is not an error because of negative acoustic modes (unconverged phonon calcs)